File: src/main/java/net/guizhanss/ultimategenerators2/implementation/items/capacitors/HiddenCapacitor.java
package net.guizhanss.ultimategenerators2.implementation.items.capacitors;

import javax.annotation.ParametersAreNonnullByDefault;

import org.bukkit.inventory.ItemStack;

import io.github.thebusybiscuit.slimefun4.api.items.ItemGroup;
import io.github.thebusybiscuit.slimefun4.api.items.SlimefunItemStack;
import io.github.thebusybiscuit.slimefun4.api.recipes.RecipeType;

public class HiddenCapacitor extends UGCapacitor {
    @ParametersAreNonnullByDefault
    public HiddenCapacitor(ItemGroup itemGroup, SlimefunItemStack item, RecipeType recipeType, ItemStack[] recipe, int capacity) {
        super(itemGroup, item, recipeType, recipe, capacity);
    }

    @ParametersAreNonnullByDefault
    public HiddenCapacitor(ItemGroup itemGroup, SlimefunItemStack item, RecipeType recipeType, ItemStack[] recipe, int capacity, int craftAmount) {
        super(itemGroup, item, recipeType, recipe, capacity, craftAmount);
    }

    @Override
    public void postRegister() {
        setHidden(true);
    }
}

File: src/main/kotlin/net/guizhanss/ultimategenerators2/implementation/items/capacitors/HiddenCapacitor.kt
package net.guizhanss.ultimategenerators2.implementation.items.capacitors

import io.github.thebusybiscuit.slimefun4.api.items.ItemGroup
import io.github.thebusybiscuit.slimefun4.api.items.SlimefunItemStack
import io.github.thebusybiscuit.slimefun4.api.recipes.RecipeType
import org.bukkit.inventory.ItemStack

open class HiddenCapacitor : UGCapacitor {
    constructor(
        itemGroup: ItemGroup,
        item: SlimefunItemStack,
        recipeType: RecipeType,
        recipe: Array<ItemStack?>,
        capacity: Int
    ) : super(
        itemGroup, item, recipeType, recipe, capacity
    )

    constructor(
        itemGroup: ItemGroup,
        item: SlimefunItemStack,
        recipeType: RecipeType,
        recipe: Array<ItemStack?>,
        capacity: Int,
        craftAmount: Int
    ) : super(
        itemGroup, item, recipeType, recipe, capacity, craftAmount
    )

    override fun postRegister() {
        isHidden = true
    }
}
